### PR TITLE
added option to set default shell globally

### LINKED ISSUE #4003: Use global default shell in `publish-crates.yml` workflow
To remove the need to repeat default shell declaration in our workflow, we should set a global default so that it is available across all jobs in https://github.com/build-trust/ockam/blob/cdf925aa2adb4439061449e5ea5f9ad774833e20/.github/workflows/publish-crates.yml#L1 
More here https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#defaults

### PR DESCRIPTION
**Pull request by RishiKumarRay**
## Current Behavior

Use global default shell in publish-crates.yml workflow
## Proposed Changes
added deafult shell option in workflow file

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [ ] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

### CODE REVIEW

File: .github/workflows/publish-crates.yml [hunk 1]
```diff
@@ -27,6 +27,9 @@ permissions:
 jobs:
   publish_crates:
     runs-on: ubuntu-20.04
+    defaults:
+      run:
+        shell: bash
```
Comment (metaclips) [inline .github/workflows/publish-crates.yml]: Let's make this global across all jobs. I think placing this right after https://github.com/build-trust/ockam/blob/cdf925aa2adb4439061449e5ea5f9ad774833e20/.github/workflows/publish-crates.yml#L23-L25 sounds okay.

**Review by metaclips — CHANGES_REQUESTED** [review]
Thanks @RishiKumarRay for an amazing PR, little changes needed and we merge.

File: .github/workflows/publish-crates.yml (hunk 1 above)
Comment (RishiKumarRay) [reply .github/workflows/publish-crates.yml]: @metaclips please check now

Comment (metaclips) [reply .github/workflows/publish-crates.yml]: Nice! Thanks!!

**Review by metaclips — DISMISSED** [review]
Approved 🚀. We should have it merged in a bit.

**Comment by metaclips** [comment]
One last bit @RishiKumarRay 

To merge PRs in our repository, we require you to accept our CLA.  To accept the CLA, please send a different pull request to our [contributors repository](https://github.com/build-trust/ockam-contributors) indicating that you accept the CLA by adding your Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file.

We also have a [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages) which is key to how we generate our changelogs.

**Comment by RishiKumarRay** [comment]
Hi @metaclips added , please check
https://github.com/build-trust/ockam-contributors/pull/172

**Comment by metaclips** [comment]
Thanks for an amazing PR @RishiKumarRay we should get it merged now.
We noticed that there was a mail mismatch, the email indicated in this PR https://github.com/build-trust/ockam-contributors/pull/172 `<rishirai760@gmail.com>` points to a different account  https://github.com/Rishirai7 we were able to use a mail recognised by GitHub https://github.com/build-trust/ockam-contributors/commit/1471b45183d28a69ad0919aec110b8c84df168d3. Can you please sign another CLA with your GitHub recognised email. Thanks.

**Review by mrinalwadhwa — DISMISSED** [review]

**Review by mrinalwadhwa — APPROVED** [review]

**Pull request merged**